common/lib/xmodule/xmodule/course_module.py

```diff
@@ -1584,6 +1592,19 @@ def can_toggle_course_pacing(self):
             return False
         return datetime.now(utc) <= self.start
 
+    def get_discussion_toggle_status(self):
+        child_statuses = [
+            child.get_discussion_toggle_status()
+            for child in self.get_children()
+        ]
+
+        if "partially_enabled" in child_statuses or ("enabled" in child_statuses and "disabled" in child_statuses):
+            return "partially_enabled"
+        elif all([status == "enabled" for status in child_statuses]):
+            return "enabled"
+        else:
+            return "disabled"
+
```

Comment:
If you use the code from I've shared below, you don't need this. The code below will work for all cases.